### ISSUE TITLE
`TokenEncryptionKey` is described as optional, but encoding fails if not specified

### ISSUE THREAD

**Issue by abjerner** [umbraco/Umbraco.AuthorizedServices#13]
According to the `README.md` the [`TokenEncryptionKey`](https://github.com/umbraco/Umbraco.AuthorizedServices#tokenencryptionkey) is optional, but trying to authenticate via the package, encoding fails as a secret key is required:

https://github.com/umbraco/Umbraco.AuthorizedServices/blob/main/src/Umbraco.AuthorizedServices/Services/Implement/AesSecretEncryptor.cs#L17

So the question is - should the README be updated? Or should the behavior be updated to align with the README?

**Comment by AndyButland** [comment]
My thinking is we should allow this to be optional, but make sure not to fail if it's not provided.  I've done that with the linked PR.  If you have time for a look and have any thoughts, would be appreciated.

**Comment by abjerner** [comment]
It takes care of the barrier for using the package, which I think is good, but I'm not super thrilled about storing unencoded tokens. Would it be an idea to use `Umbraco:CMS:Global:Id` as the secret in case a `TokenEncryptionKey` isn't provided?

And perhaps show an warning/alert in the backoffice when the "fallback" is used?

**Comment by AndyButland** [comment]
Makes sense, have updated to do that.

**Comment by AndyButland** [comment]
Thanks.  We did have this as the default at one point, but made it the optional implementation... I'm trying to think why.  Possibly related to a concern about if you were load balancing, where an encrypted value from one node wouldn't be accessile on another?  But I hadn't realised the CMS was configuring it already.

Changing the default would cause a couple of issues:

- Existing stored access tokens would no longer work.
- If you do want to use the alternate, as it stands it's a one-liner, to go the other way needs a bit more set up that has been added in the linked PR.

We are still in `0.X` experimental version stage, but because of the above I think we should leave this for now and leave it to the implementer to swap if they want.

Spelling mistake has been fixed in the PR for this issue, and I've also removed from the docs the noted need to enable data protection.
